llvm/include/llvm/Object/ELF.h

```diff
@@ -891,6 +924,39 @@ Expected<uint64_t> ELFFile<ELFT>::getDynSymtabSize() const {
 
 template <class ELFT> ELFFile<ELFT>::ELFFile(StringRef Object) : Buf(Object) {}
 
+template <class ELFT> Error ELFFile<ELFT>::readShdrZero() {
+  const Elf_Ehdr &Header = getHeader();
+
+  if ((Header.e_phnum == ELF::PN_XNUM || Header.e_shnum == 0 ||
+       Header.e_shstrndx == ELF::SHN_XINDEX) &&
+      Header.e_shoff != 0) {
+
+    // Pretend we have section 0 or sections() would call getShNum and thus
+    // become an infinite recursion
```

Comment:
```suggestion
    // become an infinite recursion.
```
Nit: comments should be full sentences, with trailing full stops.